syntax = "proto3";

option java_package = "ex.grpc";
option objc_class_prefix = "HSW";

package example;

service AuctionService {
  rpc CreateAuction (CreateAuctionRequest) returns (CreateAuctionReply) {}
  rpc GetAuctions (GetAuctionsRequest) returns (GetAuctionsReply) {}
  rpc PaymentComplete (PaymentCompleteRequest) returns (PaymentCompleteReply) {}
}

message CreateAuctionRequest {
  string auction_name = 1;
  int32 seller_id = 2;
  double price = 3;
}

message CreateAuctionReply {
  string message = 1;
}

message GetAuctionsRequest {
  int32 next_auction_id = 1;
  int32 limit = 2;
}

message Auction {
  int32 auction_id = 1;
  string auction_name = 2;
  double start_price = 3;
  string seller_name = 4;
  string start_at = 5;
  string end_at = 6;
  string status = 7;
}

message GetAuctionsReply {  
  repeated Auction auctions = 1;
  int32 next_auction_id = 2;
}

message PaymentCompleteRequest {
  int32 payment_id = 1;
}

message PaymentCompleteReply {
  string message = 1;
}